kr1/Vector.Src/SparseVector.cs

```diff
@@ -0,0 +1,142 @@
+// copyright file: https://github.com/kamendov-maxim/semester2/blob/master/LICENSE
+namespace Vector;
+
+/// <summary>
+/// Implementation of SparseVector
+/// </summary>
+public class SparseVector
+{
+    private readonly SortedDictionary<int, int> dictionary;
+    public readonly int Length;
+    public SparseVector(int[] vector)
+    {
+        ArgumentNullException.ThrowIfNull(vector);
+        dictionary = [];
+        Length = vector.Length;
+
+        int i = 0;
+        foreach (var number in vector)
+        {
+            if (number != 0)
+            {
+                dictionary[i] = number;
+            }
+            ++i;
+        }
+    }
+
+    /// <summary>
+    /// Check if vector contains only zeros
+    /// </summary>
+    /// <returns>True if there are no other numbers than zeros</returns>
+    public bool IsZeroVector()
+    {
+        return dictionary.Count == 0;
+    }
```

Comment:
```suggestion
    public bool IsZeroVector()
        => dictionary.Count == 0;
```
Более того, по традиции все методы, возвращающие состояние объекта и не требующие серьёзных вычислений (например, этот) делаются свойствами. Например, Count у библиотечного списка.